crypto/time.go

```diff
@@ -7,7 +7,9 @@ import (
 
 // UpdateTime updates cached time.
 func UpdateTime(newTime int64) {
-	pgp.latestServerTime = newTime
+	if(newTime > pgp.latestServerTime) {
+		pgp.latestServerTime = newTime
+	}
 	pgp.latestClientTime = time.Now()
```

Comment:
I would put this line inside the if as well, otherwise `getDiff()` calculation below will return some strange results.